text/0000-stack-buildpacks.md

```diff
@@ -0,0 +1,596 @@
+
+### Resolving Mixins
+
+After the detect phase, the lifecycle will merge the list of provided mixins from the following sources:
+* `stack.toml`
+* `buildpack.toml` of any stack buildpacks
+* The build plan (any `[[provides]]` tables with `mixin = true`)
+
+If any required mixins from the Build Plan (any `[[required]]` tables with `mixin = true`) are not provided, then the build will fail. Otherwise the build will continue.
+
+If a stack buildpack provides a mixin that is not required, the stack buildpack MAY pass detection. This is in contrast to a userspace buildpack providing a dependency that is not required, which fails detection. If a mixin is required for a [single stage only](https://github.com/buildpacks/rfcs/pull/109) with the `build:` or `run:` prefix, a stack buildpack may provide it for both stages without failing detection. However, it will not be included in the Buildpack Build Plan during the stage where it is not required.
+
+During the build phase, the lifecycle will create a build plan containing only the entries required during that stage (build or run).
+* If a mixin is required for "run" stage only, it will not appear in the build plan entries during build
+* If a mixin is required for "build" stage only, it will not appear in the build plan entries during extend
+
+The entries of the build plan during the build and extend phases will have a new `mixin` key. For example:
+
+```
+[[entries]]
+name = "libpq"
+mixin = true
+```
+
+A Stack Buildpack that needs to install mixins must select them from the build plan.
+
+## Caching and Restoring
+
+During the export phase, the lifecycle will store any snapshot layers created during the build phase in the cache.
+
+During the restore phase of the next build, the lifecycle will download the snapshot layer with the cache and store it as a tarball in the `<layers>` directory (i.e. it will not extract it). The `restorer` cannot extract the snapshot because it will not run with root privileges. In addition, the `restorer` may run in a different container than the build, which means changes made to the base image are not guaranteed to carry forward.
+
+During the build phase, the lifecycle will extract and apply each snapshot tarball before running stack buildpacks, but after a snapshot-baseline has been captured. This ensures that all changes from the previous snapshot are preserved even if the stack buildpack does not make any additional changes.
+
+## Rebasing an App
+
+Before a launch image is rebased, the platform must re-run the any stackpacks that were used to build the launch image against the new run-image. The image containing the stack buildpacks and the builder binary must be provided to the rebaser operation as an argument. A platform may choose to provide the same stack buildpacks and builder binary used during the build that create the launch-image being rebased, or it may provide updates versions (which may increase the risk of something failing in the rebase process).
```

Comment:
Can you please elaborate on what is the "launch image"?

Reply:
fixed in ca9b034